src/app/(pages)/groups/components/groups/index.tsx

```diff
@@ -0,0 +1,86 @@
+import { Section } from "@/components/layout";
+import { ScimReference, User } from "@/models/scim";
+import AddGroupButton from "./add-group-button";
+import JoinGroupButton from "./join-group-button";
+import GroupsTable from "./table";
+import { Suspense } from "react";
+import { auth } from "@/auth";
+import { getGroupsPage } from "@/services/groups";
+import { InputQuery } from "@/components/inputs";
+import Paginator from "@/components/paginator";
+import { Paginated } from "@/models/pagination";
+import { makeScimReferenceFromGroup } from "@/utils/scim";
+
+type GroupsProps = {
+  count: number;
+  page: number;
+  query?: string;
+  me?: User;
+}
+
+function paginateMyGroups(
+  me: User,
+  count: number,
+  startIndex: number = 1,
+  filter?: string
+) {
+  let scimGroups = me.groups ?? [];
+  if (filter) {
+    scimGroups = scimGroups.filter(g => g.display.includes(filter));
```

Comment:
`filter` can be undefined in `g.display.includes` and thus you can remove `if (filter)` at all

Reply:
Apparently this is false